pymodbus/client/asynchronous/serial.py

```diff
@@ -38,6 +38,9 @@ def _framer(cls, method):
 
         raise ParameterException("Invalid framer method requested")
 
+    def __iter__(self):
```

Comment:
Why do we have to make `AsyncModbusSerialClient` an iterable ?

Reply:
Without the __iter__ SonarCloud reported 2 bugs and asked that we use an iterable object, therefore I added __iter__

Actually sometimes I do not understand SonarCloud, I think it reports a lot of false positives.

Reply:
Thats a false positive , it assumes it requires an iterable for having the variable `loop` in the examples.
<img width="1321" alt="image" src="https://user-images.githubusercontent.com/9276974/161704127-a5881b26-842a-427e-b08c-15be45a13cb5.png">

Reply:
Please remove that, I will mark that as false positivie

Reply:
Yes it is, but I cannot find a away to disable the specific line, expected to be able to do something like #sonarcloud: disable=iter but it seems sonar cloud do not support such features.

Reply:
I will remove it, but please tell me how to mark it as a false positive.